pyfolio/utils.py

```diff
@@ -126,25 +130,39 @@ def load_portfolio_risk_factors(filepath_prefix=None):
         Risk factors timeseries.
     """
 
-    if filepath_prefix is None:
-        import pyfolio
-        filepath = os.path.join(os.path.dirname(pyfolio.__file__), 'data')
-    else:
-        filepath = filepath_prefix
-
-    factors = pd.read_csv(os.path.join(
-        filepath, 'F-F_Research_Data_Factors_daily.csv'), index_col=0)
-    mom = pd.read_csv(os.path.join(
-        filepath, 'daily_mom_factor_returns_fixed_dates2.csv'),
-                      index_col=0, parse_dates=True)
-
-    factors.index = [datetime.fromtimestamp(
-        time.mktime(time.strptime(str(t), "%Y%m%d"))) for t in factors.index]
-
-    five_factors = factors.join(mom)
-    # transform the returns from percent space to raw values (to be consistent
-    # with our portoflio returns values)
-    five_factors = five_factors / 100
+    five_factors = None
+
+    # If it's been more than two days since we updated, redownload CSVs
+    if datetime.now() - pd.to_datetime(os.path.getmtime('data/factors.h5'),
+                                       unit='s') > pd.Timedelta(days=2):
+        try:
+            umd_req = urlopen('http://mba.tuck.dartmouth.edu/page'
+                              's/faculty/ken.french/ftp/F-F_Momentum'
+                              '_Factor_daily_CSV.zip')
+            factors_req = urlopen('http://mba.tuck.dartmouth.edu/pag'
+                                  'es/faculty/ken.french/ftp/F-F_Re'
+                                  'search_Data_Factors_daily_CSV.zip')
+
+            umd_zip = zipfile.ZipFile(StringIO(umd_req.read()), 'r')
+            factors_zip = zipfile.ZipFile(StringIO(factors_req.read()),
+                                          'r')
+            umd_csv = umd_zip.read('F-F_Momentum_Factor_daily.CSV')
+            umd_csv = umd_csv.split('\r\n\r\n')[2]
+            factors_csv = factors_zip.read('F-F_Research_Data_'
+                                           'Factors_daily.CSV')
+            factors_csv = factors_csv.split('\r\n\r\n')[1]
+
+            factors = pd.DataFrame.from_csv(StringIO(factors_csv), sep=',')
+            umd = pd.DataFrame.from_csv(StringIO(umd_csv), sep=',')
+
+            five_factors = factors.join(umd)
+            five_factors = five_factors / 100
+            five_factors.to_hdf('data/factors.h5', 'df')
```

Comment:
I think this might only work if you're importing pyfolio from the source dir. Might have to do:
```python
-    if filepath_prefix is None:
-        import pyfolio
-        filepath = os.path.join(os.path.dirname(pyfolio.__file__), 'data')
-    else:
-        filepath = filepath_prefix
```
But you can try.